### ISSUE TITLE
Error: The argument type 'String' can't be assigned to the parameter type 'DiagnosticsNode'.

### ISSUE THREAD

**Issue by karnadii** [renancaraujo/photo_view#135]
**Describe the bug**
Error after upgrading to v1.5.9-pre.151 channel master, see flutter/flutter#31962 for details.

```
file:///home/karnadi/Android/flutter/.pub-cache/hosted/pub.dartlang.org/photo_view-0.3.1/lib/src/photo_view_utils.dart:169:72: Error: The argument type 'String' can't be assigned to the parameter type 'DiagnosticsNode'.
 - 'DiagnosticsNode' is from 'package:flutter/src/foundation/diagnostics.dart' ('file:///home/karnadi/Android/flutter/packages/flutter/lib/src/foundation/diagnostics.dart').
Try changing the type of the parameter, or casting the argument to 'DiagnosticsNode'.
            context: 'while dispatching notifications for $runtimeType',
                                                                       ^
file:///home/karnadi/Android/flutter/.pub-cache/hosted/pub.dartlang.org/photo_view-0.3.1/lib/src/photo_view_utils.dart:170:35: Error: The argument type 'Null Function(StringBuffer)' can't be assigned to the parameter type 'Iterable<DiagnosticsNode> Function()'.
 - 'StringBuffer' is from 'dart:core'.                                  
 - 'Iterable' is from 'dart:core'.                                      
 - 'DiagnosticsNode' is from 'package:flutter/src/foundation/diagnostics.dart' ('file:///home/karnadi/Android/flutter/packages/flutter/lib/src/foundation/diagnostics.dart').
Try changing the type of the parameter, or casting the argument to 'Iterable<DiagnosticsNode> Function()'.
            informationCollector: (StringBuffer information) {   
```

**Comment by renancaraujo** [comment]
Seems like a Breaking change in Flutter's master. Thanks for reporting it out.

**Comment by yashshah7197** [comment]
I've refactored the `photo_view_utils.dart` file to use the new FlutterError objects to fix this issue for users on the master channel. Also updated version constraints to use the master channel for now.

However, since this will not allow users on the stable channel to use any new version published, we could update the README to include a section for installation versions for stable and master channel users. 

#136

**Comment by keitoro** [comment]
how do I apply this fix?

**Comment by renancaraujo** [comment]
@yashshah7197 Thanks for your contribution. I made a comment. After that matter solved, I would be happy to merge it. 

@keitoro after that, you could use, temporarily, photo_view directly from repo's master.

**Comment by keitoro** [comment]
it is working again, just a shame they didn't roll this out step by step.

**Comment by yashshah7197** [comment]
@renancaraujo Thanks for the great package haha! 💯 Replied to your comment. Are you gonna publish an updated version on pub ASAP or just leave it on master for now?

**Comment by renancaraujo** [comment]
For now, I will keep on master until figure out a way to not break for beta and stable channel users. If it is possible, the next version will be `0.3.2`, if not, `0.4.0`.

**Comment by renancaraujo** [comment]
Solved compat issues on #137. Launching change as `0.3.2`
